package.json

```diff
@@ -103,6 +113,18 @@
 			}
 		],
 		"menus": {
+			"explorer/context": [
+                {
+                    "command": "liberty.dev.add.project",
+                    "when": "explorerResourceIsRoot",
```

Comment:
In the following scenario, I manually added the "finish" project to the Liberty dashboard. Since it was already added, is it possible to only show the "Remove project from Liberty Dashboard" when I right-click on the project root?
<img width="572" alt="image" src="https://user-images.githubusercontent.com/26146482/193850252-567af0de-ba34-44fd-a3d4-bc9ec04ab2f5.png">

Could consider as a future enhancement depending on how complex this is.

Reply:
It is no possible.  AFAIK, we are unable to run any code when the context menu is being displayed.

Reply:
I think we may be able to run code using the "when" clause: https://code.visualstudio.com/api/references/when-clause-contexts
but we would have to add a custom when clause context: https://code.visualstudio.com/api/references/when-clause-contexts#add-a-custom-when-clause-context. 

Seems like this could be scoped to a separate issue marked as an enhancement? What do you think?

Reply:
No, it is not possible.  The context value must be simple types, not function.